ibis/expr/types/tvf.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
```

Comment:
`tvf.py` isn't the best file name:

- It's an acronym that not everyone will be familiar with
- These functions are table valued functions, but they're also specific to time windowing.

Since we don't support any other table valued functions (and may not plan to, at least under that name), perhaps it'd make more sense to name/group this functionality based on what it does rather than the function signature (`f(table: Table) -> Table)`). Perhaps `window_by.py` or `temporal_windows.py`?

Same goes for `ibis/expr/operations/tvf.py`.